### PR TITLE
Add E2E tests for broker syncer

### PR DESCRIPTION
**Pull request by tpantelis**

### CODE REVIEW

File: scripts/kind-e2e/e2e.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+with_context cluster2 deploy_broker
+
+
+export BROKER_K8S_REMOTENAMESPACE=broker
+
+export BROKER_K8S_APISERVER=$(kubectl --context=cluster2 -n default get endpoints kubernetes -o jsonpath="{.subsets[0].addresses[0].ip}:{.subsets[0].ports[?(@.name=='https')].port}")
+
+export BROKER_K8S_CA=$(kubectl --context=cluster2 -n ${BROKER_K8S_REMOTENAMESPACE} get secrets -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='${BROKER_K8S_REMOTENAMESPACE}-client')].data['ca\.crt']}")
+
+export BROKER_K8S_APISERVERTOKEN=$(kubectl --context=cluster2 -n ${BROKER_K8S_REMOTENAMESPACE} get secrets -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='${BROKER_K8S_REMOTENAMESPACE}-client')].data.token}"|base64 --decode)
```
Comment (mkolesnik) [inline scripts/kind-e2e/e2e.sh:55]: Are these used? I find no mention of them in the PR besides exporting them here

File: scripts/kind-e2e/e2e.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+function cleanup {
+    "${SCRIPTS_DIR}"/cleanup.sh
+}
+
+### Main ###
+
+declare_kubeconfig
+
+with_context cluster1 deploy_crds
+with_context cluster2 deploy_crds
```
[the remark this answers is not in the document]
Comment (tpantelis) [reply scripts/kind-e2e/e2e.sh:45]: yup

File: scripts/kind-e2e/e2e.sh [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    go test -v -args -v=2 -logtostderr=true -ginkgo.v -ginkgo.randomizeAllSpecs \
+        -dp-context cluster1 -dp-context cluster2 \
+        -ginkgo.reportFile ${DAPPER_OUTPUT}/e2e-junit.xml 2>&1 | \
+        tee ${DAPPER_OUTPUT}/e2e-tests.log
+}
+
+function cleanup {
+    "${SCRIPTS_DIR}"/cleanup.sh
+}
+
```
[the remark this answers is not in the document]
Comment (tpantelis) [reply scripts/kind-e2e/e2e.sh:39]: yes - I've since removed it.

File: scripts/kind-e2e/e2e.sh [hunk 4]
```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+
+## Process command line flags ##
+
+source /usr/share/shflags/shflags
+FLAGS "$@" || exit $?
+eval set -- "${FLAGS_ARGV}"
+
```
[the remark this answers is not in the document]
Comment (tpantelis) [reply scripts/kind-e2e/e2e.sh:8]: yes - I've since removed it.

File: scripts/deploy [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    kubectl apply -f ${DAPPER_SOURCE}/test/e2e/deploy/crds.yaml
+}
+
+declare_kubeconfig
+
+run_all_clusters deploy_crds
+with_context "${clusters[1]}" deploy_broker
+
+
+
```
Comment (dfarrell07) [inline scripts/deploy:22]:
```suggestion
```

Extra newlines.

Comment (tpantelis) [reply scripts/deploy]: Removed

File: test/e2e/deploy/crds.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+kind: CustomResourceDefinition
+metadata:
+  name: exportedtoasters.admiral.submariner.io
+spec:
+  group: admiral.submariner.io
+  version: v1
+  names:
+    kind: ExportedToaster
+    plural: exportedtoasters
+  scope: Namespaced
```
Comment (skitt) [inline test/e2e/deploy/crds.yaml:24]:
```suggestion
  scope: Namespaced
```

(missing newline, not worth fixing on its own but I have other suggestions elsewhere).

File: test/e2e/syncer/broker.go [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+)
+
+var _ = Describe("[syncer] Broker bi-directional syncer tests", func() {
+	Context("with a specific source namespace", testWithSpecificSourceNamespace)
+	Context("with source namespace all", testWithSourceNamespaceAll)
+	Context("with local transform", testWithLocalTransform)
+})
+
+func testWithSpecificSourceNamespace() {
+	t := newTestDiver()
```
Comment (skitt) [inline test/e2e/syncer/broker.go:31]:
```suggestion
	t := newTestDriver()
```

File: test/e2e/syncer/broker.go [hunk 8]
```diff
@@ -0,0 +1,328 @@
...
+
+	BeforeEach(func() {
+		t.localSourceNamespace = t.framework.Namespace
+	})
+
+	t.bidirectionalSyncTests()
+}
+
+func testWithSourceNamespaceAll() {
+	newTestDiver().bidirectionalSyncTests()
```
Comment (skitt) [inline test/e2e/syncer/broker.go:41]:
```suggestion
	newTestDriver().bidirectionalSyncTests()
```

File: test/e2e/syncer/broker.go [hunk 9]
```diff
@@ -0,0 +1,328 @@
...
+}
+
+func testWithLocalTransform() {
+	t := newTestDiver()
```
Comment (skitt) [inline test/e2e/syncer/broker.go:45]:
```suggestion
	t := newTestDriver()
```

File: test/e2e/syncer/broker.go [hunk 10]
```diff
@@ -0,0 +1,328 @@
...
+
+func (t *testDriver) createToaster(cluster framework.ClusterIndex) *testV1.Toaster {
+	namespace := t.framework.Namespace
+	toaster := &testV1.Toaster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      framework.TestContext.ClusterIDs[cluster] + "-toaster",
+			Namespace: namespace,
+		},
+		Spec: testV1.ToasterSpec{
+			Manufacturer: "Cuisinart",
```
Comment (skitt) [inline test/e2e/syncer/broker.go:179]: Heh, perhaps we can get a sponsorship deal ;-).

**Pull request merged**